### DISCUSSION TITLE
How to get Invoke-DbaQuery respect the date-time format from SQL Server

### DISCUSSION THREAD

**Discussion by Shashahan** [dataplat/dbatools#9442] [Q&A]
Dear all,

date and time formats in SQL Server follow this convention:
https://learn.microsoft.com/en-us/sql/t-sql/functions/date-and-time-data-types-and-functions-transact-sql?view=sql-server-ver16

Data type | Format
-|-
time | hh:mm:ss[.nnnnnnn] 
date | YYYY-MM-DD
smalldatetime|YYYY-MM-DD hh:mm:ss
datetime(|YYYY-MM-DD hh:mm:ss[.nnn]
datetime2|YYYY-MM-DD hh:mm:ss[.nnnnnnn]
datetimeoffset|YYYY-MM-DD hh:mm:ss[.nnnnnnn] 

But Invoke-DbaQuery changes the datetime format to the currently used format of the powershell session

How can I change this behaviour to have the datetime format be the same as the source?

Thank you very much and kind regards,
Olaf

**Comment by niphlod** [comment]
care to post an example of what you're trying to do ? I'm pretty sure we forward the underlying datetime object, so you're the one responsible to convert it into your desired format if you want to pass to a string representation of the underlying datetime value.

**Comment by Shashahan** [comment]
Powershell was the culprit, not dbatools. Becasuse of the powershell session is set to use the US format all dates where displayed with format mm/dd/yyyy and the Export-CSV also used this format.

Solution was to convert the date into a string with the query.
